homeassistant/components/whois/sensor.py

```diff
@@ -39,33 +41,52 @@ def setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the WHOIS sensor."""
-    domain = config[CONF_DOMAIN]
-    name = config[CONF_NAME]
-
+    LOGGER.warning(
+        "Configuration of the Whois platform in YAML is deprecated and will be "
+        "removed in Home Assistant 2022.4; Your existing configuration "
+        "has been imported into the UI automatically and can be safely removed "
+        "from your configuration.yaml file"
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={CONF_DOMAIN: config[CONF_DOMAIN], CONF_NAME: config[CONF_NAME]},
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the platform from config_entry."""
+    domain = entry.data[CONF_DOMAIN]
     try:
-        if "expiration_date" in whois.whois(domain):
-            add_entities([WhoisSensor(name, domain)], True)
-        else:
-            LOGGER.error(
-                "WHOIS lookup for %s didn't contain an expiration date", domain
-            )
-            return
+        info = await hass.async_add_executor_job(whois.whois, domain)
     except whois.BaseException as ex:  # pylint: disable=broad-except
         LOGGER.error("Exception %s occurred during WHOIS lookup for %s", ex, domain)
         return
 
+    if "expiration_date" not in info:
+        LOGGER.error("WHOIS lookup for %s didn't contain an expiration date", domain)
+        return
+
+    async_add_entities([WhoisSensor(domain)], True)
+
 
 class WhoisSensor(SensorEntity):
     """Implementation of a WHOIS sensor."""
 
     _attr_icon = "mdi:calendar-clock"
     _attr_native_unit_of_measurement = TIME_DAYS
 
-    def __init__(self, name: str, domain: str) -> None:
+    def __init__(self, domain: str) -> None:
         """Initialize the sensor."""
+        self._attr_name = domain
```

Comment:
You are probably planning to do it in a follow-up PR, but since you are dropping the name from the config entry, does it not make sense to add the unique_id to the entity directly in this PR to be able to rename it in the UI?

Reply:
Yeah, that is the idea :)
I wanted to move that into the PR that adds more sensors/sensor entity descriptions to ensure a stable unique ID from the get-go.